src/main/java/edu/cornell/kfs/fp/batch/service/impl/AccountingXmlDocumentDownloadAttachmentServiceImpl.java

```diff
@@ -72,7 +76,8 @@ public Attachment createAttachmentFromBackupLink(Document document,
                 return downloadByteArrayFromWebResource(accountingXmlDocumentBackupLink.getFileName(), request);
             } else {
                 LOG.error("downloadByteArray, the group code isn't valid for the link URL");
-                throw new IOException("The group code isn't valid for the link URL");
+                throw new ValidationException("The group code: " + accountingXmlDocumentBackupLink.getCredentialGroupCode()
+                                + " isn't valid for the link URL: " + accountingXmlDocumentBackupLink.getLinkUrl());
             }
         } catch (URISyntaxException e) {
             throw new IOException(e.getMessage());
```

Comment:
Should this be changed to throw a ValidationException instead? Otherwise, the calling code won't print the message of the IOException when it catches it.

Reply:
This IO exception will be caught on line 65 and then logged:
LOG.error("createAttachmentFromBackupLink, Unable to download attachment: " + accountingXmlDocumentBackupLink.getLinkUrl(), e);

Reply:
The line after that log statement will just throw a ValidationException with a generic "Unable to download attachment" message. Is that enough information for someone reading the report file? Or is a more descriptive message needed to indicate that the URL's syntax was invalid?